src/tests.rs

```diff
@@ -59,6 +60,19 @@ fn parse_file() {
 }
 
 #[test]
+fn serialize_and_read_file() {
+    let mut path = TempDir::new("test_rm_tempdir").unwrap().into_path();
```

Comment:
http://doc.rust-lang.org/tempdir/tempdir/struct.TempDir.html#method.into_path
> This discards the wrapper so that the automatic deletion of the temporary directory is prevented.

This doesn’t sound like what we want here. Rather, assign the result of `unwrap` to a variable, use `.path()` on it, and when that variable goes out of scope at the end of the test the temporary directory will be cleaned up (removed) automatically.